src/App.tsx

```diff
@@ -26,70 +27,122 @@ type ReorderOptions = {
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+    if (sortType === SortType.LENGTH) {
+      visibleGoods.sort((a, b) => a.length - b.length);
+    }
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return visibleGoods;
+  };
+
+  sortAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortLength = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reverse = () => {
+    this.setState({ isReversed: !this.state.isReversed });
+  };
+
+  reset = () => {
+    this.setState({ isReversed: false });
+    this.setState({ sortType: SortType.NONE });
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    let visibleGoods = [...goodsFromServer];
+
+    if (sortType === SortType.ALPHABET) {
+      visibleGoods = this.getReorderedGoods(goodsFromServer,
+        { sortType: SortType.ALPHABET, isReversed });
+    } else if (sortType === SortType.LENGTH) {
+      visibleGoods = this.getReorderedGoods(goodsFromServer,
+        { sortType: SortType.LENGTH, isReversed });
+    }
+
+    if (isReversed) {
+      visibleGoods = this.getReorderedGoods(goodsFromServer,
+        { sortType, isReversed: true });
+    }
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={sortType === SortType.ALPHABET
+              ? 'button is-info'
+              : 'button is-info is-light'}
+            onClick={this.sortAlphabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={sortType === SortType.LENGTH
+              ? 'button is-success'
+              : 'button is-success is-light'}
+            onClick={this.sortLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={isReversed === true
+              ? 'button is-warning'
+              : 'button is-warning is-light'}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          {(isReversed || sortType !== SortType.NONE) && (
+            <>
+              <button
+                type="button"
+                className="button is-danger is-light"
+                onClick={this.reset}
+              >
+                Reset
+              </button>
+            </>
+          )}
+
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
```

Comment:
here we have now `<ul>` inside `<ul>` think one is enough so this can be removed :)